OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -144,7 +144,12 @@ void INotifyCreated.Created(Actor self)
 			conditionManager = self.Trait<ConditionManager>();
 		}
 
-		static int GetWeight(Actor a) { return a.Info.TraitInfo<PassengerInfo>().Weight; }
+		public int GetWeight(Actor a) { return GetPassengerTrait(a).Info.Weight; }
+
+		public Passenger GetPassengerTrait(Actor a)
```

Comment:
Cargo isn't really an appropriate place for this.  I'd prefer that you moved this inline (see my comment below), but failing that this would be better as a static method on Passenger.